docs/DeveloperGuide.md

```diff
@@ -51,30 +51,63 @@ Step 5. The string at the 0th index is then used in a switch statement, where ea
+When the user launches the app, the main program will initialize a `TopicsStorage` object and call the `load` method 
+which will return a `TopicList` object. The following sequence diagram shows how the load operation works:
 
 ![TopicsStorage load](./images/TopicsStorage_load.png)
 
+### Class Component 
+
+#### Option and OptionList Class 
+
+The `Option` and `OptionList` classes implements the `Displayable` and `DisplayableList` interfaces respectively. 
+The `Option` object stores one option of a question while the `OptionList` object stores all 4 options of the same 
+question. The class diagram below illustrates the structure of both classes. 
+
+![Option_and_OptionList_Class](./images/Option.png)
+
+### User Interface 
+
+#### Implementation 
+
+The `Ui` class handles all the interactions with the users. It reads the input from the users and prints out replies to 
+the users. An example is provided below to illustrate how the `Ui` class prints out the starting page of the quiz for 
+the user to comprehend. 
+
+![Ui_Printing_Start_Quiz_Page](./images/PrintQuizStartPage.png)
+
 ## Product scope
 ### Target user profile
 
 CS2113/T Students
 
 ### Value proposition
 
-Help CS2113/T students learn and understand software engineering and OOP principles through a gamified platform and enhance their learning experience. Consolidate key concepts for easy revision.
+Help CS2113/T students learn and understand software engineering and OOP principles through a gamified platform and 
+enhance their learning experience. Consolidate key concepts for easy revision.
 
 ## User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
 |v1.0|new user|answer given questions|start testing myself immediately|
+|v1.0|new user|quickly see what the software has to offer|understand how to use the app|
+|v1.0|new user|answer given questions|start testing myself immediately|
+|v1.0|long-time user|get different questions each time|repeatedly test my understanding for the particular topic|
+|v1.0|busy user|test myself on concepts using short, targeted quizzes|confirm my understanding of concepts|
+|v1.0|user|get a hint for the question in the quiz|I can think about the question from a different angle|
+|v1.0|user|see what the available topics are|navigate around the app effectively|
+|v1.0|user|select the number of questions to do in the quiz|manage the workload and time spent on the quiz|
+|v2.0|slow but hardworking user|see the explanations provided in the quiz|learn from my mistakes and revise|
+|v2.0|busy, lazy user|take note of key concepts|refer to it easily at a later time|
+|v2.0|frequent disorganized user|view the percentage of error in each topic|tell how well I understand the content|
```

Comment:
Thanks for adding this in! Looks good!

Reply:
I think the first one is repeated now because I added it previously but thanks for adding all of the rest

Reply:
Oh ok, I have changed this too! Thank you!